internal/node/reverse_proxy/handler.go

```diff
@@ -0,0 +1,146 @@
+package reverse_proxy
+
+import (
+	"fmt"
+	"io/fs"
+	"net"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/eagraf/habitat-new/core/state/node"
+	"github.com/eagraf/habitat-new/internal/frontend"
+	"github.com/eagraf/habitat-new/internal/node/config"
+	"github.com/rs/zerolog/log"
+)
+
+func getHandlerFromRule(rule *node.ReverseProxyRule, nodeConfig *config.NodeConfig) (http.Handler, error) {
+	switch rule.Type {
+	case node.ProxyRuleRedirect:
+		return getRedirectHandler(rule)
+	case node.ProxyRuleFileServer:
+		return getFileServerHandler(rule, WithBasePath(nodeConfig.WebBundlePath()))
+
+	case node.ProxyRuleEmbeddedFrontend:
```

Comment:
what's the difference between embedded frontend and file server? seems like they're both file servers but embedded frontend is special?

Reply:
The frontend isn't sitting on the host machine's file system, instead, when compiling the node binary we embed the entire built frontend bundle into the binary, so that we can serve it directly. This is only used when we aren't in development mode, and is useful because it means there's no setup step where we have to first write out the frontend to a directory on the machine after the initial binary has been installed.

Added some comments explaining this as well